llama_index/callbacks/base.py

```diff
@@ -70,40 +74,52 @@ class CallbackManager(BaseCallbackHandler, ABC):
     Args:
         handlers (List[BaseCallbackHandler]): list of handlers to use.
 
+    Usage:
+        with callback_manager.event(CBEventType.QUERY) as event:
+            event.on_start(payload={key, val})
+            ...
+            event.on_end(payload={key, val})
+
     """
 
     def __init__(self, handlers: List[BaseCallbackHandler]):
         """Initialize the manager with a list of handlers."""
         self.handlers = handlers
         self._trace_map: Dict[str, List[str]] = defaultdict(list)
-        self._trace_event_stack: List[str] = [BASE_TRACE_EVENT]
+        self._trace_event_stack: ContextVar[List[str]] = ContextVar(

```

Comment:
Saw this while reading docs on context var, thoughts? 

```
Important: Context Variables should be created at the top module level and never in closures. Context objects hold strong references to context variables which prevents context variables from being properly garbage collected.
```

Reply:
Hmm, I can try moving it to a global. I had it like that at some point, but need to confirm that it works still